File: Backend/Models/ancVisit.js
import mongoose from 'mongoose';

const ancVisitSchema = new mongoose.Schema({
    visitNumber:{
        type:String,
        required: true
    },
    visitDate:{
        type: Date,
        required: true,
        default: Date.now
    },
    gestationAge:{
        gestWeeks:{
            type: Number,
            required: true
        },
        gestDays:{
            type: Number,
            required: true
        }
    },
    weightKilos:{
        type: Number,
        required: true
    },
    bloodPressure:{
        systolic:{
            type: Number,
            required: true
        },
        diastolic:{
            type: Number,
            required: true
        }
    },
    fundalHeight:{
        type: Number,
        required: true
    },
    muac:{
        type: Number,
        required: true
    },
    complaints:{
        type: String,
        requrd: true,
        default: 'None'
    },
    medicationGiven:{
        type: String,
        required: true,
        default: 'None'
    },
    nextAppointment:{
        type: Date,
        required:true
    }
});

const ancVisit = mongoose.model('ancVisit', ancVisitSchema);
export default ancVisit;

File: api/Models/AncVisit.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace KemriApi.Models
{
    public class ancVisit
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("visitNumber")]
        public string visitNumber { get; set; } = string.Empty;

        [BsonElement("visitDate")]
        public DateTime visitDate { get; set; } = DateTime.UtcNow;

        [BsonElement("gestationAge")]
        public GestationAgeInfo gestationAge { get; set; } = new();

        [BsonElement("weightKilos")]
        public double weightKilos { get; set; }

        [BsonElement("bloodPressure")]
        public BloodPressureInfo bloodPressure { get; set; } = new();

        [BsonElement("fundalHeight")]
        public double fundalHeight { get; set; }

        [BsonElement("muac")]
        public double muac { get; set; }

        [BsonElement("complaints")]
        public string complaints { get; set; } = "None";

        [BsonElement("medicationGiven")]
        public string medicationGiven { get; set; } = "None";

        [BsonElement("nextAppointment")]
        public DateTime nextAppointment { get; set; }
    }

    public class GestationAgeInfo
    {
        [BsonElement("gestWeeks")]
        public int gestWeeks { get; set; }

        [BsonElement("gestDays")]
        public int gestDays { get; set; }
    }

    public class BloodPressureInfo
    {
        [BsonElement("systolic")]
        public int systolic { get; set; }

        [BsonElement("diastolic")]
        public int diastolic { get; set; }
    }
}
